ProcessLib/LIE/SmallDeformation/SmallDeformationProcess.cpp

```diff
@@ -96,6 +100,73 @@ SmallDeformationProcess<DisplacementDim>::SmallDeformationProcess(
             *fracture_prop.get());
     }
 
+    // set branches
+    for (std::size_t i=0; i<vec_branch_nodeID_matIDs.size(); i++)
+    {
+        auto master_matId = vec_branch_nodeID_matIDs[i].second[0];
+        auto slave_matId = vec_branch_nodeID_matIDs[i].second[1];
+        auto& master_frac =
+            *_process_data._vec_fracture_property
+                 [_process_data._map_materialID_to_fractureID[master_matId]];
+        auto& slave_frac =
+            *_process_data._vec_fracture_property
+                 [_process_data._map_materialID_to_fractureID[slave_matId]];
+
+        BranchProperty* branch = new BranchProperty();
+        setBranchProperty(*mesh.getNode(vec_branch_nodeID_matIDs[i].first),
+                          master_frac, slave_frac, *branch);
+
+        master_frac.branches_master.emplace_back(branch);
+
+        BranchProperty* branch2 = new BranchProperty();
```

Comment:
In the new code `new` is not welcome (unless you have to as in copying the MeshLib's nodes). I also didn't see the reason to store pointers in the `*.branches_master/slave` vectors instead of the branch objects.
In any case this is a memory leak.

Reply:
Because the number of branches varies depending on problem,  i store them as a vector of its pointers. i am not  sure if I understand your question.